torch/csrc/distributed/c10d/init.cpp

```diff
@@ -2010,10 +2010,8 @@ communication mechanism.
           py::arg("world_size"));
 
   auto processGroup =
-      py::class_<
-          ::c10d::ProcessGroup,
-          c10::intrusive_ptr<::c10d::ProcessGroup>,
-          ::c10d::PyProcessGroup>(module, "ProcessGroup",
```

Comment:
We need to keep the ::c10d::PyProcessGroup which is a trampoline so you can use Python classes as ProcessGroups in C++.

Might need to change the `intrusive_ptr_no_gil_destructor_class_` signature or add a new template variant to support the 3 arg version